lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3986,34 +3986,72 @@ LogicalResult ConvertAtenOp<AtenClampOp>::matchAndRewrite(
 
   IntegerAttr min_int, max_int;
   FloatAttr min_fp, max_fp;
-  if (op.getMin().getType().isa<Torch::FloatType>()) {
-    double fp_min, fp_max;
-    if (!matchPattern(op.getMin(), m_TorchConstantFloat(&fp_min)))
-      return rewriter.notifyMatchFailure(
-          op, "unimplemented: value `fp_min` should be a torch constant float");
 
-    if (!matchPattern(op.getMax(), m_TorchConstantFloat(&fp_max)))
-      return rewriter.notifyMatchFailure(
-          op, "unimplemented: value `fp_max` should be a torch constant float");
+  auto getValAttr = [&](Value operand, bool isFloat, IntegerAttr &intAttr,
```

Comment:
There's a lot of repetitive control flow here. I think we can avoid a lot of it by doing

```c++
  IntegerAttr min_int = rewriter.getI64IntegerAttr(std::numeric_limits<int64_t>::min());
  IntegerAttr max_int = rewriter.getI64IntegerAttr(std::numeric_limits<int64_t>::max());
  FloatAttr min_fp = rewriter.getF32FloatAttr(std::numeric_limits<float>::lowest());
  FloatAttr max_fp = rewriter.getF32FloatAttr(std::numeric_limits<float>::max());

  auto getValAttr = [&](Value operand, IntegerAttr &intAttr,
                        FloatAttr &fpAttr) -> LogicalResult {
    double valFloat;
    int64_t valInt;
    if (matchPattern(operand, m_TorchConstantFloat(&valFloat))) {
      intAttr = rewriter.getI64IntegerAttr(static_cast<int64_t>(valFloat));
      fpAttr = rewriter.getF32FloatAttr(static_cast<float>(valFloat));
    } else if (matchPattern(operand, m_TorchConstantInt(&valInt))) {
      intAttr = rewriter.getI64IntegerAttr(valInt);
      fpAttr = rewriter.getF32FloatAttr(static_cast<float>(valInt));
    } else {
      return failure();
    }
    return success();
  };

  LogicalResult minAttrResult = getValAttr(op.getMin(), min_int, min_fp);
  LogicalResult maxAttrResult = getValAttr(op.getMax(), max_int, max_fp);
  if (failed(minAttrResult) && failed(maxAttrResult)) {
    return rewriter.notifyMatchFailure(
        op, "either `min` or `max` should be a torch constant");
  }
  if (failed(minAttrResult) && checkNotNone(rewriter, op, op.getMin())) {
    return rewriter.notifyMatchFailure(op, "min attr should be a torch constant");
  }
  if (failed(maxAttrResult) && checkNotNone(rewriter, op, op.getMax())) {
    return rewriter.notifyMatchFailure(op, "max attr should be a torch constant");
  }
```

Reply:
This looks a lot cleaner! Thanks for the suggestion!